x-pack/solutions/security/plugins/security_solution/public/data_view_picker/components/data_view_picker/index.tsx

```diff
@@ -0,0 +1,131 @@
+  const closeFieldEditor = useRef<() => void | undefined>();
+
+  // TODO: this should be disabled for the default data views probably, eg. `security-solution-default`
+  // const canEditDataView =
+  // Boolean(dataViewEditor?.userPermissions.editDataView()) || !dataView.isPersisted();
+  const canEditDataView = true;
+
+  const { dataView } = useDataView(props.scope);
+
+  const dataViewId = dataView?.id;
+
+  const createNewDataView = useCallback(() => {
+    closeDataViewEditor.current = dataViewEditor.openEditor({
+      onSave: async (newDataView) => {
+        dispatch(shared.actions.addDataView(newDataView));
+        selectDataView({ id: newDataView.id, scope: [props.scope] });
+      },
+      allowAdHocDataView: true,
+    });
+  }, [dataViewEditor, dispatch, props.scope, selectDataView]);
+
+  const onFieldEdited = useCallback(() => {}, []);
+
+  const editField = useMemo(() => {
+    if (!canEditDataView) {
+      return;
+    }
+    return async (fieldName?: string, _uiAction: 'edit' | 'add' = 'edit') => {
+      if (!dataViewId) {
+        return;
+      }
+
+      const dataViewInstance = await data.dataViews.get(dataViewId);
+      closeFieldEditor.current = await dataViewFieldEditor.openEditor({
+        ctx: {
+          dataView: dataViewInstance,
+        },
+        fieldName,
+        onSave: async () => {
+          onFieldEdited();
+        },
+      });
+    };
+  }, [canEditDataView, dataViewId, data.dataViews, dataViewFieldEditor, onFieldEdited]);
+
+  const addField = useMemo(
+    () => (canEditDataView && editField ? () => editField(undefined, 'add') : undefined),
+    [editField, canEditDataView]
+  );
+
+  const handleChangeDataView = useCallback(
+    (id: string) => {
+      selectDataView({ id, scope: [props.scope] });
+    },
+    [props.scope, selectDataView]
+  );
+
+  const handleEditDataView = useCallback(() => {}, []);
+
+  const triggerConfig = useMemo(() => {
+    if (dataView.id === DEFAULT_SECURITY_SOLUTION_DATA_VIEW_ID) {
+      return {
+        label: 'Default Security Data View',
+      };
+    }
+
+    return {
+      label: dataView?.name || dataView?.id || 'Data view',
+    };
+  }, [dataView]);
+
+  const { adhocDataViews: adhocDataViewSpecs, dataViews } = useSelector(sharedStateSelector);
+
+  const managedDataViews = useMemo(() => {
+    const managed: DataViewListItem[] = dataViews.map((spec) => ({
+      id: spec.id ?? '',
+      title: spec.title ?? '',
+      name: spec.name,
+    }));
+
+    return managed;
+  }, [dataViews]);
+
+  const adhocDataViews = useMemo(() => {
+    return adhocDataViewSpecs.map((spec) => new DataView({ spec, fieldFormats }));
+  }, [adhocDataViewSpecs, fieldFormats]);
+
+  return (
+    <UnifiedDataViewPicker
+      currentDataViewId={dataViewId || DEFAULT_SECURITY_SOLUTION_DATA_VIEW_ID}
+      trigger={triggerConfig}
+      onChangeDataView={handleChangeDataView}
+      onEditDataView={handleEditDataView}
```

Comment:
updating data view (i.e. add/remove indices) is not automatically reflected in timeline. Requires a refresh or changing between data views to see updated results

https://github.com/user-attachments/assets/274e5d78-09d8-4def-9ec6-5baf198b0f1c

Reply:
Nice catch!

Reply:
so this is implemented now @christineweng @michaelolo24 , cant say that about field modification yet but working on it